### PR TITLE
Fail in prompter when console is not interactive

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
There's no point in using the prompter when the console is not interactive.
CLI should still be usable even if the console is not interactive so we should provide a way for every command to bypass using the prompter one way or another.
Part of [#297090](http://teampulse.telerik.com/view#item/297090)'s implementation
Merge along with https://github.com/telerik/mobile-cli-lib/pull/383

### CODE REVIEW

File: lib/commands/cloud-projects.ts [hunk 1]
```diff
@@ -27,31 +27,40 @@ export class CloudListProjectsCommand implements ICommand {
...
-		let headers =  ["#", "Project Name"];
-		let data = list.map((proj: Server.IWorkspaceItemData, index: number) => [(++index).toString(), proj.Name]);
+	private printList(names: string[], config: {isProject: boolean, solutionName?: string}): void {
+		let isProject = config.isProject;
+		let headers =  ["#", `${isProject ? 'Project' : 'Solution'} name`];
+		let data = names.map((name: string, index: number) => [(++index).toString(), name]);
 		let table = commonHelpers.createTable(headers, data);
-		this.$logger.out(`Projects for ${solutionName} solution:`);
+		if (isProject) {
+			this.$logger.out(`Projects for ${config.solutionName} solution:`);
```
Comment (rosen-vladimirov) [inline lib/commands/cloud-projects.ts]: it is possible to have `config.isProject` set to true without `config.solutionName` - in this case you will print `Projects for undefined solution`
Is this expected?

Comment (Mitko-Kerezov) [inline lib/commands/cloud-projects.ts]: fixed!

File: docs/man_pages/project/creation/cloud.md [hunk 2]
```diff
@@ -3,17 +3,20 @@ cloud
...
-List solutions and projects | `$ appbuilder cloud`
-List projects in a solution | `$ appbuilder cloud <Solution Name or Index>`
+List all solutions | `$ appbuilder cloud --all`
+List projects in a solution | `$ appbuilder cloud [<Solution Name or Index>]`
 Export projects | `$ appbuilder cloud export [<Solution Name or Index>] [<Project Name or Index>]`
 
 Lists all solutions and projects associated with your Telerik Platform account.
 
+### Options
+* `--all` - Lists all solutions with index and name. You need to use this option if you are running a non-interactive console.
```
Comment (rosen-vladimirov) [inline docs/man_pages/project/creation/cloud.md]: Do you need to use this option if console is not interactive?

File: test/cloud-projects.ts [hunk 3]
```diff
@@ -436,5 +437,42 @@ describe("cloud project commands", () => {
...
+
+				assert.isTrue(sln1Index !== -1);
+				assert.isTrue(sln2Index !== -1);
+				assert.isTrue(sln3Index !== -1);
+			});
+			
+			it("when --all option provided in interactive console", () => {
+				testInjector = createTestInjector("Sln1", "BlankProj");
+				let logger = testInjector.resolve("logger");
+				testInjector.register("options", {all: true});
```
Comment (rosen-vladimirov) [inline test/cloud-projects.ts]: as options is already registered in testInjector, it is better to use
```TypeScript
let opts = testInjector.resolve("options");
opts.all = true;
```

File: docs/man_pages/project/creation/cloud.md [hunk 4]
```diff
@@ -3,17 +3,20 @@ cloud
...
+
 ### Attributes
 
 * `export` - Extends the `cloud` command. Exports a selected project from the cloud and initializes it for development in the Telerik AppBuilder CLI. You must run this command in a directory that does not contain a project.
-* `<Solution Name or Index>` is the name of the solution as listed by `$ appbuilder cloud` or as it appears in the Telerik AppBuilder in-browser client or the Telerik AppBuilder Windows client.
-* `<Project Name or Index>` is the name or the index of project, relative to its parent solution, as listed by `appbuilder cloud` or as it appears in the Telerik AppBuilder in-browser client or the Telerik AppBuilder Windows client.
+* `<Solution Name or Index>` is the name of the solution as listed by `$ appbuilder cloud --all` or as it appears in the Telerik AppBuilder in-browser client or the Telerik AppBuilder Windows client. You need to set a solution if you are running a non-interactive console.
+* `<Project Name or Index>` is the name or the index of project, relative to its parent solution, as listed by `$ appbuilder cloud --all` or as it appears in the Telerik AppBuilder in-browser client or the Telerik AppBuilder Windows client.
```
Comment (rosen-vladimirov) [inline docs/man_pages/project/creation/cloud.md]: does --all option lists projects?

**Comment by rosen-vladimirov** [comment]
:+1: , but please ping @ikoevska for the messages

**Comment by Mitko-Kerezov** [comment]
Already did : )

**Comment by Mitko-Kerezov** [comment]
run ci

**Comment by Mitko-Kerezov** [comment]
run ci

**Comment by Mitko-Kerezov** [comment]
run ci

**Comment by Mitko-Kerezov** [comment]
run ci

**Pull request merged**